cqrs-postgres/src/db_wrapper.rs

```diff
@@ -0,0 +1,177 @@
+use crate::util::Sequence;
+use cqrs_core::{CqrsError, EventNumber, RawEvent, Since};
+use num_traits::ToPrimitive;
+use postgres::{rows::Row, types::ToSql};
+use r2d2::{Pool, PooledConnection};
+use r2d2_postgres::PostgresConnectionManager;
+use std::{error, fmt, sync::Arc};
+
+#[derive(Debug, Clone)]
+pub enum DbError {
+    Format(Arc<dyn CqrsError>),
+    Pool(Arc<dyn CqrsError>),
+    Postgres(Arc<dyn CqrsError>),
+    React(Arc<dyn CqrsError>),
+}
+
+impl DbError {
+    pub fn format(err: impl CqrsError) -> Self {
+        DbError::Format(Arc::new(err))
+    }
+
+    pub fn pool(err: impl CqrsError) -> Self {
+        DbError::Pool(Arc::new(err))
+    }
+
+    pub fn postgres(err: impl CqrsError) -> Self {
+        DbError::Postgres(Arc::new(err))
+    }
+
+    pub fn react(err: impl CqrsError) -> Self {
+        DbError::React(Arc::new(err))
+    }
+}
+
+impl From<r2d2::Error> for DbError {
+    fn from(err: r2d2::Error) -> Self {
+        DbError::Pool(Arc::new(err))
+    }
+}
+
+impl From<postgres::Error> for DbError {
+    fn from(err: postgres::Error) -> Self {
+        DbError::Postgres(Arc::new(err))
+    }
+}
+
+impl From<fmt::Error> for DbError {
+    fn from(err: fmt::Error) -> Self {
+        DbError::Format(Arc::new(err))
+    }
+}
+
+impl fmt::Display for DbError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            DbError::Format(ref err) => write!(f, "{}", err),
+            DbError::Pool(ref err) => write!(f, "{}", err),
```

Comment:
Add a prefix indicating that there was an error running the reaction.

Reply:
Done